GeoLib/AnalyticalGeometry.cpp

```diff
@@ -466,13 +466,16 @@ std::vector<MathLib::Point3d> lineSegmentIntersect2d(
     }
 
     auto isCollinearPointOntoLineSegment = [](MathLib::Point3d const& a,
-        MathLib::Point3d const& b, MathLib::Point3d const& c)
-        {
-            double const t((c[0]-a[0])/(b[0]-a[0]));
-            if (0.0 <= t && t <= 1.0)
-                return true;
-            return false;
-        };
+                                              MathLib::Point3d const& b,
+                                              MathLib::Point3d const& c) {
+        if (b[0] - a[0] == 0)
+            OGS_FATAL(
+                "isCollinearPointOntoLineSegment is about to divide by "
+                "zero: a[0] = %g, b[0] = %g, and b[0] - a[0] = %g",
+                a[0], b[0], b[0] - a[0]);
```

Comment:
Can't the algorithm be fixed by trying any of the other coordinates, i.e. `b[1] - a[1]` or `b[2] - a[2]`?